admin-part/src/main/kotlin/Coverage.kt

```diff
@@ -162,21 +164,31 @@ private fun Sequence<Session>.bundlesByTests(
     context: CoverContext,
     cache: AtomicCache<TypedTest, BundleCounter>?
 ): Map<TypedTest, BundleCounter> = takeIf { it.any() }?.run {
+    val testsWithEmptyBundle = testsWithEmptyBundle(cache)
     groupBy(Session::testType).map { (testType, sessions) ->
         sessions.asSequence().flatten()
             .groupBy { TypedTest(it.testName, testType) }
             .filterNot { cache?.map?.containsKey(it.key) ?: false }
-            .map { it.key to it.value.asSequence().bundle(context) }
-            .toMap(mutableMapOf())
+            .mapValuesTo(testsWithEmptyBundle) { it.value.asSequence().bundle(context) }
     }.reduce { m1, m2 ->
         m1.apply { putAll(m2) }
     }.let { mutableMap ->
-        val finalizedTests = flatMap { map -> map.testStats.keys }
+        val finalizedTests = flatMap { it.testStats.keys }
         val testsAddToCache = mutableMap.filterKeys { finalizedTests.contains(it) }
         mutableMap + (cache?.putAll(testsAddToCache) ?: emptyMap())
     }
 } ?: emptyMap()
 
+private fun Sequence<Session>.testsWithEmptyBundle(
+    cache: AtomicCache<TypedTest, BundleCounter>?,
+    emptyBundle: BundleCounter = BundleCounter("")
```

Comment:
If you want to add bundle as parameter then rename parameter and method. But I don't understand for what)

Reply:
The only purpose that I pursued, adding bundle as a parameter was that: the value in the map should be a reference to a single object , and not creating a unique object for each key , which is not unique in fact.
The original code was as follows
```
    private fun Sequence<Session>.testsWithEmptyBundle(
    cache: AtomicCache<TypedTest, BundleCounter>?,
): MutableMap<TypedTest, BundleCounter> = BundleCounter("").let { emptyBundle ->
    flatMap { session ->
        session.tests.asSequence().filterNot { cache?.map?.containsKey(it) ?: false }
    }.associateWithTo(mutableMapOf()) {
        emptyBundle
    }
}
```
but @LikhanovMaxim asked to move bundle to params

Reply:
Then you need to rename the parameter and the method, because you can pass a non-empty bundle.